common/app/components/core/HtmlComponent.scala

```diff
@@ -0,0 +1,48 @@
+package components.core
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.parser.Parser
+import play.twirl.api.Html
+import scala.collection.JavaConversions._
+
+trait HtmlComponent {
+
+  def html: Html
+  def componentCss: Seq[Css]
+
+  def render(): Html = {
+    val document: Document = Jsoup.parse(html.toString, "", Parser.xmlParser()) //Provide an XML Parser so Jsoup doesn't automatically add <html>, <head> and <body> tags
```

Comment:
Yep. Ideally I would prefer not to use `Jsoup` altogether. It is ok for a quick prototype but I have been burn by `Jsoup` a few time in the past. 
Anyway your advice is very good